### ISSUE TITLE
 Support for Dotty 3.0.0-M1

### ISSUE THREAD

**Issue by Sciss** [etorreborre/specs2#877]
Hi there, you may have seen it already; the artifacts of Dotty (Scala) 3.0.0.-M1 have been published, [see blog](https://dotty.epfl.ch/blog/2020/11/09/scala3-m1.html). so it would be great if specs2 could release `_3.0` artifacts as well.

**Comment by etorreborre** [comment]
Yes I saw that. Unfortunately the compiler is currently choking on my code with something like
```
[error] ## Exception when compiling 59 sources to /Users/etorreborre/projects/specs2/specs2-five/tests/target/scala-3.0.0-M1/test-classes
[error] java.lang.AssertionError: assertion failed: position not set for $1$.set(minTestsOk = 50, $1$.set$default$2, $1$.set$default$3, $1$.set$default$4
[error]   ,
[error] $1$.set$default$5, $1$.set$default$6, $1$.set$default$7):
[error]   org.specs2.scalacheck.ScalaCheckFunction1[Int, org.scalacheck.Prop] # -1 of class http://dotty.tools.dotc.ast.Trees$Typed in core/shared/src/main/scala/org/specs2/specification/create/S2StringContext.scala
[error] dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
[error] http://dotty.tools.dotc.typer.Typer$.assertPositioned(Typer.scala:64)
[error] http://dotty.tools.dotc.typer.Typer.typed(Typer.scala:2584)
[error] http://dotty.tools.dotc.typer.Typer.typed(Typer.scala:2593)
```
And I haven't had time to investigate yet. Hopefully I can fix / work-around this during this week-end.

**Comment by etorreborre** [comment]
There was a work-around indeed, so I published specs2 with version `5.0.0-3.0.0-M1-bin-20201029-1f433d5-NIGHTLY`.

I had to use Scala 3.0.0 `3.0.0-M1-bin-20201029-1f433d5-NIGHTLY` because `3.0.0-M1` just doesn't work out of the box.
